store/sqlstore/channel_store.go

```diff
@@ -416,6 +421,260 @@ func (s SqlChannelStore) createIndexesIfNotExists() {
 	s.CreateIndexIfNotExists("idx_channels_scheme_id", "Channels", "SchemeId")
 }
 
+func (s SqlChannelStore) MigrateSidebarCategories(fromTeamId, fromUserId string) (map[string]string, *model.AppError) {
```

Comment:
This method is quite complicated, I know we don't normally do this, but in cases like this I think it would be worthwhile adding a comment explaining what it does at a high level. Based on my understanding, it goes over every `TeamMember` and creates 3 default categories for it: for Favourites, Direct Messages and Channels. Is that correct?